src/services/AccountService.js

```diff
@@ -35,72 +35,90 @@ export default class AccountService {
+        return AccountSecureStorage.removeAccount(id, network);
     }
 
     /**
      * Renames account by it's id
      */
-    static async renameAccount(id: string, newName: string): string {
-        const allAccounts = await AccountSecureStorage.getAllAccounts();
-        allAccounts.forEach(account => {
-            if (account.id === id) {
-                account.name = newName;
-            }
-        });
-        await AccountSecureStorage.saveAccounts(allAccounts);
-        return allAccounts;
+    static async renameAccount(id: string, newName: string, network: AppNetworkType): string {
+        const allAccounts = await AccountSecureStorage.getAllAccountsByNetwork(network);
+        const account = allAccounts.find(account => account.id === id);
+        account.name = newName;
+        await AccountSecureStorage.updateAccount(account);
+        return AccountSecureStorage.getAllAccountsByNetwork(network);
     }
 
     /**
      * Updates persistent del request sent
      */
-    static async updateDelegatedHarvestingInfo(id: string, isPersistentDelReqSent: boolean, harvestingNode: string): string {
-        const allAccounts = await AccountSecureStorage.getAllAccounts();
-        allAccounts.forEach(account => {
-            if (account.id === id) {
-                account.isPersistentDelReqSent = isPersistentDelReqSent;
-                account.harvestingNode = harvestingNode;
-            }
-        });
-        await AccountSecureStorage.saveAccounts(allAccounts);
-        return allAccounts;
+    static async updateDelegatedHarvestingInfo(id: string, isPersistentDelReqSent: boolean, harvestingNode: string, network: AppNetworkType): string {
+        const allAccounts = await AccountSecureStorage.getAllAccountsByNetwork(network);
+        const account = allAccounts.find(account => account.id === id);
+        if (!account) return;
+        account.isPersistentDelReqSent = isPersistentDelReqSent;
+        account.harvestingNode = harvestingNode;
+        await AccountSecureStorage.updateAccount(account);
+        return AccountSecureStorage.getAllAccountsByNetwork(network);
     }
 
     /**
      * Creates an account from a mnemonic
      * @param mnemonic
      * @param index
      * @param name
+     * @param network
      * @returns {AccountModel}
      */
-    static createFromMnemonicAndIndex(mnemonic: string, index: number, name: string): AccountModel {
+    static createFromMnemonicAndIndex(mnemonic: string, index: number, name: string, network: AppNetworkType): AccountModel {
         const mnemonicPassPhrase = new MnemonicPassPhrase(mnemonic);
         const seed = mnemonicPassPhrase.toSeed().toString('hex');
         const extKey = ExtendedKey.createFromSeed(seed);
         const wallet = new Wallet(extKey);
-        const path = `m/44'/4343'/${index}'/0'/0'`;
+        const path = `m/44'/${network === 'testnet' ? '1' : '4343'}'/${index}'/0'/0'`;
```

Comment:
IMO paths should be accessible from one place: separate class / constants / config